ufl/argument.py

```diff
@@ -19,27 +19,25 @@
 from ufl.split_functions import split
 from ufl.finiteelement import FiniteElementBase
 from ufl.domain import default_domain
+from ufl.form import BaseForm
 from ufl.functionspace import AbstractFunctionSpace, FunctionSpace, MixedFunctionSpace
+from ufl.duals import is_primal, is_dual
 
 # Export list for ufl.classes (TODO: not actually classes: drop? these are in ufl.*)
 __all_classes__ = ["TestFunction", "TrialFunction", "TestFunctions", "TrialFunctions"]
 
 
 # --- Class representing an argument (basis function) in a form ---
 
-@ufl_type()
-class Argument(FormArgument):
+class BaseArgument(object):
```

Comment:
Bikeshedding, I kind of prefer `AbstractArgument` over `BaseArgument` to more concretely indicate that the class should not be instantiated directly.